src/main/java/net/minecraftforge/fluids/FluidTank.java

```diff
@@ -175,11 +198,28 @@ public FluidStack drain(int maxDrain, boolean doDrain)
                 fluid = null;
             }
 
+            onContentsChanged();
+
             if (tile != null)
             {
                 FluidEvent.fireEvent(new FluidEvent.FluidDrainingEvent(fluid, tile.getWorld(), tile.getPos(), this, drained));
             }
         }
         return stack;
     }
+
+    public boolean canFill(Fluid fluid)
```

Comment:
canFill/canDrain are just for filtering basically. They defines the types of fluids that the tanks allows to flow in and out.